### DISCUSSION TITLE
Embedder crashing at node::InitializeNodeWithArgs with C0000005 (access violation) on Windows

### DISCUSSION THREAD

**Discussion by giilbert** [nodejs/help#4908] [Q&A]
### Version

v17.6.1-pre (built from source from https://github.com/nodejs/node/tree/v17.x-staging)

### Platform

Microsoft Windows NT 10.0.19044.0 x64

### What steps will reproduce the bug?

I'm trying to build node into a static library and link it to a project with CMake.

1. Build node from https://github.com/nodejs/node/tree/v17.x-staging with flags ` --fully-static` & `--enable-static`.  I used MS Build Tools 2019.
2. Build the [embedding example](https://github.com/nodejs/node/blob/v17.x-staging/test/embedding/embedtest.cc) with options:  
  - Compiler flags: `MT`
  - Definitions: `DV8_COMPRESS_POINTERS`, `D_ITERATOR_DEBUG_LEVEL=0`
  - Includes: `v8`, `node`, `uv`
  - Linking: `v8_monolith.lib` (v 9.9-lkgr), `node.lib` (v17.x-staging), `dbghelp.lib`, `Winmm.lib`
  - Link options: `/ignore:4099`
3. Build with CMake with no issue! Put a copy of node.exe generated during the build at the output directory and run it `test.exe console.log(42)`.
4. Crashes with exit code -1073741819 [hex C0000005]

### What is the expected behavior?

The program would not crash and 42 gets outputted to the console.

### What do you see instead?

- Program silently crashes with exit code -1073741819 [hex C0000005] (access violation).
- It doesn't make it to the error handling after `node::InitializeNodeWithArgs`.

Using the debugger, I found that it crashes at
```
int exit_code = node::InitializeNodeWithArgs(&args, &exec_args, &errors);
```
with `Exception 0xc0000005 encountered at address 0x51dfefe: User-mode data execution prevention (DEP) violation at location 0x051dfefe`
`-858993460` also got stored in exit_code variable

Did I build node wrong? If so, how do I build it into a .lib correctly?

### Additional information

CMakeLists.txt - https://gist.github.com/giilbert/db130af69dce695af2f40fb83c87352e

**Comment by bnoordhuis** [comment]
I've converted this to a discussion because this is likely not a bug in node.js itself.
